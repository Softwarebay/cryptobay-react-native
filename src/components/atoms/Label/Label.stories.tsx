import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import React from 'react';

import StorybookView from '../../../../storybook/StorybookView';

import { Label } from './Label';

storiesOf('Atoms/Label', module)
  .addDecorator(getStory => <StorybookView>{getStory()}</StorybookView>)
  .add('Price Label', () => <Label value={text('value', '19.2371 BNB')} />);
