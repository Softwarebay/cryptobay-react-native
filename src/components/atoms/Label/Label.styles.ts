import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  label: {
    fontFamily: 'Poppins',
    fontSize: 16,
    lineHeight: 24,
    color: '#FFFFFF',
    fontWeight: '600',
  },
});
