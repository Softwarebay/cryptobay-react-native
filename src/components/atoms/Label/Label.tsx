import React from 'react';
import { Text } from 'react-native';

import { styles } from './Label.styles';

type LabelProps = {
  value: string;
  testID?: string;
};

export const Label: React.FC<LabelProps> = ({ value, testID }: LabelProps) => {
  return (
    <Text testID={testID} style={styles.label}>
      {value}
    </Text>
  );
};
