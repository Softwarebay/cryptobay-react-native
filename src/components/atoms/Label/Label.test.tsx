import { render, screen } from '@testing-library/react-native';
import React from 'react';
import '@testing-library/jest-native';
import renderer from 'react-test-renderer';

import 'react-native';
import { Label } from './Label';

describe('Label Atom', () => {
  test('Component Snapshot', () => {
    const tree = renderer.create(<Label value="Next" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('ensure label value renders correctly', () => {
    render(<Label value="Next" testID="atom-label" />);
    expect(screen.getByTestId('atom-label')).toHaveTextContent('Next');
  });
});
