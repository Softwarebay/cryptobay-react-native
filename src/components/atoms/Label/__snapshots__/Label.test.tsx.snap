// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label Atom Component Snapshot 1`] = `
<Text
  style={
    Object {
      "color": "#FFFFFF",
      "fontFamily": "Poppins",
      "fontSize": 16,
      "fontWeight": "600",
      "lineHeight": 24,
    }
  }
>
  Next
</Text>
`;
